lib/buildkite/test_collector/tracer.rb

```diff
@@ -4,6 +4,13 @@
 
 module Buildkite::TestCollector
   class Tracer
+    # https://github.com/buildkite/test-collector-ruby/issues/131
+    class MonotonicTime
+      def self.call
+        Process.method(:clock_gettime).call Process::CLOCK_MONOTONIC
```

Comment:
I'm a bit nervous about this. Have you tested if it works when Process.clock_gettime is stubbed? In the [ruby-timeout implementation](https://github.com/ruby/timeout/pull/18/files#diff-8ad4c142bc670fe498191f02758e39480e7b06e3ff80be9abfc1d73337d35a83R140) they assign the reference to a constant:

![image](https://user-images.githubusercontent.com/543166/176132973-956a6949-08d5-451a-8f9a-ebece17c6da2.png)

That constant will be evaluated & stored when the class is defined. If anything mocks the Process.clock_gettime method in future it won't impact the constant and the original method can be called.

In your implementation I think a mocking library would replace `Process.method(:clock_gettime)` and it wouldn't work as desired (although without testing I'm honestly not sure).

Reply:
Yea I tested the `Process.method(:clock_gettime)` will be mocked and changed back to store in constant instead c70c9a7. Thanks!